pymc_marketing/clv/models/basic.py

```diff
@@ -78,31 +131,120 @@ def _fit_MAP(self, **kwargs):
+            If the inference data that is loaded doesn't match with the model.
+        Examples
+        --------
+        >>> class MyModel(ModelBuilder):
+        >>>     ...
+        >>> name = './mymodel.nc'
+        >>> imported_model = MyModel.load(name)
+        """
+        filepath = Path(str(fname))
+        idata = az.from_netcdf(filepath)
+        dataset = idata.fit_data.to_dataframe()
+
+        model = cls(
+            dataset,
+            model_config=json.loads(idata.attrs["model_config"]),  # type: ignore
+            sampler_config=json.loads(idata.attrs["sampler_config"]),
+        )
+        model.idata = idata
+
+        model.build_model()
+        # All previously used data is in idata.
+
+        return model
+
+    @staticmethod
+    def _check_prior_ndim(prior, ndim=0):
+        if prior.ndim != ndim:
+            raise ValueError(
+                f"Prior variable {prior} must be have {ndim} ndims, but it has {prior.ndim} ndims."
+            )
+
+    def create_distribution_from_prior(self, name: str, **kwargs) -> TensorVariable:
+        try:
+            prior = getattr(pm, name).dist(**kwargs)
+        except AttributeError:
+            raise ValueError(f"Distribution {name} does not exist in PyMC")
+        return prior
+
+    @staticmethod
+    def _process_priors(
+        *priors: TensorVariable, check_ndim: bool = True
+    ) -> Tuple[TensorVariable, ...]:
+        """Check that each prior variable is unique and attach `str_repr` method."""
+        if len(priors) != len(set(priors)):
+            raise ValueError("Prior variables must be unique")
+        # Related to https://github.com/pymc-devs/pymc/issues/6311
+        for prior in priors:
+            if check_ndim:
+                CLVModel._check_prior_ndim(prior)
+            prior.str_repr = types.MethodType(str_for_dist, prior)  # type: ignore
+        return priors
+
+    @property
+    def default_sampler_config(self) -> Dict:
+        return {}
 
     @property
-    def fit_result(self):
-        if self._fit_result is None:
+    def prior_predictive(self) -> az.InferenceData:
+        if self.idata is None or "prior_predictive" not in self.idata:
             raise RuntimeError("The model hasn't been fit yet, call .fit() first")
-        return self._fit_result
+        return self.idata["prior_predictive"]
+
+    @property
+    def fit_result(self) -> Dataset:
+        if self.idata is None or "posterior" not in self.idata:
+            raise RuntimeError("The model hasn't been fit yet, call .fit() first")
+        return self.idata["posterior"]
 
     @fit_result.setter
-    def fit_result(self, res):
-        if self._fit_result is not None:
+    def fit_result(self, res: az.InferenceData) -> None:
+        if self.idata is None:
+            self.idata = res
+        elif "posterior" in self.idata:
             warnings.warn("Overriding pre-existing fit_result")
-        self._fit_result = res
+            self.idata.posterior = res
```

Comment:
This will fail if a user called `prior_predictive` already, in which case there should be an `idata` already. Related to my comment above

Reply:
added a case that doesn't need 'posterior' to be in the idata, covered with test